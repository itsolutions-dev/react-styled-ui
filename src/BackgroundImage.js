import styledComponent, { css } from './styledComponent';

const BackgroundImage = styledComponent('div', css`
  background-size: cover;
  background-position: center;
  height: 0;
  padding-bottom: ${props => props.ratio !== undefined ? props.ratio * 100 : '100'}%;
  background-image: url(${props => props.src});
`);

export default BackgroundImage;
