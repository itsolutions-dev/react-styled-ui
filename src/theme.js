export default {
  radius: 8,
  spacing: {
    margin: 8,
    padding: 8,
    iconSize: 24,
    borderSize: 1,
    shadowSize: 1,
  },
  font: {
    size: 16,
    weight: 200,
    lineHeight: 18,
    family: 'Roboto, sans-serif',
  },
  h1: {
    size: 32,
    weight: 900,
    lineHeight: 64,
  },
  h2: {
    size: 24,
    weight: 700,
    lineHeight: 32,
  },
  h3: {
    size: 18,
    weight: 500,
    lineHeight: 24,
  },
  selected: {
    size: 0,
    weight: 10,
    lineHeight: 0,
  },
  hover: {
    size: 0,
    weight: 10,
    lineHeight: 0,
  },
  transitions: {
    in: undefined,
    out: undefined,
  },
  palette: {
    default: 'green',
    defaultText: 'white',
    defaultDisabled: undefined,
    defaultHover: undefined,
    defaultSelected: undefined,
    primary: 'red',
    primaryText: 'white',
    primaryDisabled: undefined,
    primaryHover: undefined,
    primarySelected: undefined,
    secondary: 'blue',
    secondaryText: 'white',
    secondaryDisabled: undefined,
    secondaryHover: undefined,
    secondarySelected: undefined,
    disabled: 'lightgray',
    shadow: 'black',
  },
};
