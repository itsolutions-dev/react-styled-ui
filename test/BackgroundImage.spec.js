import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import 'styled-components-test-utils/lib/jest';
import theme from '../src/theme';
import BackgroundImage from '../src/BackgroundImage';

describe('BackgroundImage', () => {
  test('should render a BackgroundImage', () => {
    const component = ReactTestRenderer.create(<BackgroundImage theme={theme} />);
    expect(component).toBeDefined();
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('should have a div tag', () => {
    const component = ReactTestRenderer.create(<BackgroundImage theme={theme} />).toJSON();
    expect(component.type).toEqual('div');
  });

  test('should handle a src prop', () => {
    const component = ReactTestRenderer.create(
      <BackgroundImage
        theme={theme}
        src="http://localhost:8080/image.png"
      />,
    );
    expect(component).toHaveStyleRule('background-image', 'url(http://localhost:8080/image.png)');
  });

  test('should have a default ratio', () => {
    const component = ReactTestRenderer.create(
      <BackgroundImage
        theme={theme}
      />,
    );
    expect(component).toHaveStyleRule('padding-bottom', '100%');
  });

  test('should handle a ratio prop', () => {
    const component = ReactTestRenderer.create(
      <BackgroundImage
        theme={theme}
        ratio={1 / 2}
      />,
    );
    expect(component).toHaveStyleRule('padding-bottom', '50%');
  });
});
