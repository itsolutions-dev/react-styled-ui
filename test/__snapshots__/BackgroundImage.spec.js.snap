// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackgroundImage should render a BackgroundImage 1`] = `
.c0 {
  background-size: cover;
  background-position: center;
  height: 0;
  padding-bottom: 100%;
  background-image: url();
}

<div
  className="c0"
/>
`;
